### ISSUE TITLE
current code is not reentrant when multipeer.py is included as an API

### ISSUE THREAD

**Issue by mithrendal** [mikaelho/multipeer#3]
hi @mikaelho 

the code is not reentrant if I do extract the demo chat code from multipeer.py into a separate file lets say demochat.py. 

demochat.py runs fine the first time, but when you exit it and run it again it will not find any peers. 

the curios thing is that when you run the demochat inside the multipeer.py then you can exit and rerun it as often you like it will always find the peers. 

In other words, running the multipeer.py  itself is completely reentrant. But calling the multipeer API from another file is not. 

 Do you have any ideas ? 

Maybe it has something to do with the delegates, which you only create once??

**Comment by mikaelho** [comment]
Thanks, you are probably right. As an imported module, it is only run once. I will modify.

**Comment by mithrendal** [comment]
I uploaded the external demo code file in my fork. just in case... ;-)

https://github.com/mithrendal/multipeer/blob/master/chat.py

**Comment by mikaelho** [comment]
Fixed in 25d20db. Also duplicated the chat demo in a separate file for testing. Thanks for reporting this.

**Comment by mithrendal** [comment]
I did just some testing on the fixed version. It works!  Very cool. ;-)
